engine.py

```diff
@@ -124,12 +123,35 @@ def __maya_exiting_callback(watcher):
         watcher.stop_watching()
 
 
+def maya_scene_path():
```

Comment:
Added this method to better replicate what Pymel was doing.
I'm not sure if it is entirely necessary to go to these lengths to check the current file name.
Our workfiles hook only uses `cmds.file(q=1, sceneName=1` to do it.